src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Numerics.Tensors
+{
+    public interface ITensor<TSelf, T>
+        : IEnumerable<T>,
+          IEquatable<TSelf>,
+          IEqualityOperators<TSelf, TSelf, bool>
+        where TSelf : ITensor<TSelf, T>
+        where T : IEquatable<T>
+    {
+        // TODO: Determine if we can implement `IEqualityOperators<TSelf, T, bool>`.
+        // It looks like C#/.NET currently hits limitations here as it believes TSelf and T could be the same type
+        // Ideally we could annotate it such that they cannot be the same type and no conflicts would exist
+
+        static TSelf? Empty { get; }
+
+        bool IsEmpty { get; }
+        bool IsPinned { get; }
+        int Rank { get; }
+        ReadOnlySpan<nint> Strides { get; }
```

Comment:
Just noting that we may need to consider how this impacts implementing the interface on a struct type, since it can't be marked `UnscopedRef`